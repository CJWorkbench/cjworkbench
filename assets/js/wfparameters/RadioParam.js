// Simple wrapper over HTML <select>
import React from 'react'
import PropTypes from 'prop-types'

export default class RadioParam extends React.PureComponent {
  onChange = (evt) => {
    var idx = evt.target.value;
    this.props.onChange(idx);
  }

  isChecked = (idx) => {
    // Cast to string because this.props.selectedIdx comes in both as int and string, for some reason.
    return idx === this.props.selectedIdx.toString()
  }

  render() {
    var items = this.props.items.split('|');
    var itemDivs = items.map( (name, idx) => {
      const strIdx = idx.toString()
      return (
        <label className='t-d-gray content-3'>
          <input
            className='radio-button'
            type='radio'
            value={strIdx}
            checked={this.isChecked(strIdx)}
            onChange={this.onChange}
            disabled={this.props.isReadOnly}
          />
          <span className="button"></span>
          {name}
        </label>
      )});

    return (
      <div className='button-group'>
        {itemDivs}
      </div>
    );
  }
}

RadioParam.propTypes = {
  name:         PropTypes.string.isRequired,
  items:        PropTypes.string,  // like 'Apple|Banana|Kitten'
  selectedIdx:  PropTypes.number,
  isReadOnly:   PropTypes.bool,
  onChange:     PropTypes.func     // called with index of selected item
};
