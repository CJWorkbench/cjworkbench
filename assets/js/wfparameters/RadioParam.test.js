import React from 'react'
import { mount } from 'enzyme'

import RadioParam from './RadioParam';

describe('RadioButtons', () => {
  const wrapper = (extraProps = {}) => mount(
    <RadioParam
      name='radio-buttons'
      items='Apple|Kittens|Banana'
      selectedIdx={0}
      onChange={jest.fn()}
      {...extraProps}
    />
  )
  it('renders correctly', () => {
    const w = wrapper({isReadOnly: true})
    expect(wrapper).toMatchSnapshot();
  });

  it('radio renders number of buttons correctly', () => {
    const w = wrapper({isReadOnly: false});
    expect(w.find('input[type="radio"]')).toHaveLength(3)
  });

  it('returns correct value when clicked', () => {
    const w = wrapper({isReadOnly: false});
    w.find('input[value="2"]').simulate('change')
    expect(w.prop('onChange')).toHaveBeenCalledWith('2')
  });

  it('should be disabled when read only', () => {
    let items = 'Apple|Kittens|Banana'.split('|')
    const w = wrapper({isReadOnly: true});
    for (let item in items) {
      let button = w.find(('input[value="' + item + '"]'))
      expect(button.prop('disabled')).toEqual(true)
    }
  });
})
