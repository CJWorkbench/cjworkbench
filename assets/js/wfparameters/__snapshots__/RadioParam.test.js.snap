// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtons renders correctly 1`] = `[Function]`;
